### DISCUSSION TITLE
cannot execute binary file: Exec format error (cloudflared)

### DISCUSSION THREAD

**Discussion by jintor** [Homebrew/discussions#2433] [Linux]
Hi,

I did `brew install cloudflare/cloudflare/cloudflared`

and saw `==> Installing cloudflared from cloudflare/cloudflare`

when I do `cloudflared` I get `-bash: /home/linuxbrew/.linuxbrew/bin/cloudflared: cannot execute binary file: Exec format error` 

How to debug this ?

I'm using Almalinux 8.4

**Comment by gromgit** [comment]
Cloudflare's [cloudflared](https://github.com/cloudflare/homebrew-cloudflare/blob/master/cloudflared.rb) formula, in its entirety:
```
class Cloudflared < Formula
  desc 'Cloudflare Tunnel'
  homepage 'https://developers.cloudflare.com/cloudflare-one/connections/connect-apps'
  url 'https://packages.argotunnel.com/dl/cloudflared-2021.11.0-darwin-amd64.tgz'
  sha256 'f0cd1bd0f6ae7f775ab157c3ae757af7f75ee0423a939b1d39f38ca5b6a36ce4'
  version '2021.11.0'
  def install
    bin.install 'cloudflared'
  end
end
```
It installs from a _macOS_ tarball, not a Linux one. You might want to [raise an issue](https://github.com/cloudflare/homebrew-cloudflare/issues) with them; Homebrew has supported OS/architecture-specific "guards" in formulae for a long time now, so formulae like this really shouldn't exist.
